diff -urN libonvif-2.0.9/onvif-util/src/onvif-util.cpp libonvif-2.0.9.mod/onvif-util/src/onvif-util.cpp
--- libonvif-2.0.9/onvif-util/src/onvif-util.cpp	2024-07-30 17:09:30.239963895 +0900
+++ libonvif-2.0.9.mod/onvif-util/src/onvif-util.cpp	2024-07-30 16:49:33.945594610 +0900
@@ -42,7 +42,8 @@
              { "user",       required_argument, NULL,      'u'},
              { "password",   required_argument, NULL,      'p'},
              { "all",        no_argument,       NULL,      'a'},
-             { "dump",        no_argument,       NULL,      'd'},
+             { "dump",       no_argument,       NULL,      'd'},
+             { "reboot",     no_argument,       NULL,      'r'},
 			 { "safe_off",   no_argument,       NULL,      's'},
              { "help",       required_argument, NULL,      'h'},
 			 { "version",    no_argument,       NULL,      'v'},
@@ -191,9 +192,10 @@
 	bool safe_mode = true;
 	bool time_sync = false;
 	bool dump_mode = false;
+	bool reboot_mode = false;
 
 	int ch;
-	while ((ch = getopt_long(argc, argv, "u:p:adhsvt", longopts, NULL)) != -1) {
+	while ((ch = getopt_long(argc, argv, "u:p:adhrsvt", longopts, NULL)) != -1) {
 		switch (ch) {
             case 'u':
 				username = optarg;
@@ -207,6 +209,9 @@
 			case 'd':
 				dump_mode = true;
 				break;
+			case 'r':
+				reboot_mode = true;
+				break;
 			case 'h':
 				usage();
 				showHelp();
@@ -290,6 +295,9 @@
 		if (dump_mode) {
 			strncpy(kybd_buf, "dump", 4);
 		}
+		else if (reboot_mode) {
+			strncpy(kybd_buf, "reboot", 6);
+		}
 		else {
 			fgets(kybd_buf, 128, stdin);
 		}
@@ -689,7 +697,12 @@
 			else if (args[0] == "reboot") {
 				std::cout << "  Are you sure you want to reboot?  Type yes to confirm\n" << std::endl;
 				memset(kybd_buf, 0, 128);
-				fgets(kybd_buf, 128, stdin);
+				if (reboot_mode) {
+					strncpy(kybd_buf, "yes", 3);
+				}
+				else {
+					fgets(kybd_buf, 128, stdin);
+				}
 				kybd_buf[strcspn(kybd_buf, "\r\n")] = 0;
 				std::string reply(kybd_buf);
 				if (reply == "yes") {
@@ -700,6 +713,9 @@
 				else {
 					std::cout << "  Confirmation not received, reboot cancelled\n" << std::endl;
 				}
+				if (reboot_mode) {
+					exit(0);
+				}
 			}
 			else if (args[0] == "sync_time") {
 				if (args.size() > 1) {
