diff -urN a/target/linux/mediatek/image/mt7981.mk b/target/linux/mediatek/image/mt7981.mk
--- a/target/linux/mediatek/image/mt7981.mk	2024-06-24 16:58:23.909342419 +0900
+++ b/target/linux/mediatek/image/mt7981.mk	2024-08-31 14:55:23.111949596 +0900
@@ -48,7 +48,7 @@
   DEVICE_MODEL := mt7981-spim-nand-gsw
   DEVICE_DTS := mt7981-spim-nand-gsw
   DEVICE_DTS_DIR := $(DTS_DIR)/mediatek
-  SUPPORTED_DEVICES := mediatek,mt7981-rfb,ubi
+  SUPPORTED_DEVICES := mediatek,mt7981-spim-snand-gsw-rfb
   UBINIZE_OPTS := -E 5
   BLOCKSIZE := 128k
   PAGESIZE := 2048
