diff -urN a/arch/arm64/boot/dts/mediatek/mt7981-spim-nand-gsw.dts b/arch/arm64/boot/dts/mediatek/mt7981-spim-nand-gsw.dts
--- a/arch/arm64/boot/dts/mediatek/mt7981-spim-nand-gsw.dts	2025-04-10 10:23:08.664054844 +0000
+++ b/arch/arm64/boot/dts/mediatek/mt7981-spim-nand-gsw.dts	2025-04-12 10:01:42.921549678 +0000
@@ -64,13 +64,18 @@
 			};
 
 			partition@580000 {
+				label = "reserved";
+				reg = <0x580000 0x0080000>;
+			};
+
+			partition@600000 {
 				label = "ubi";
-				reg = <0x580000 0x6E00000>;
+				reg = <0x600000 0x5A00000>;
 			};
 
 			partition@7400000 {
-				label = "LogA";
-				reg = <0x7400000 0x400000>;
+				label = "log";
+				reg = <0x6000000 0x1000000>;
 			};
 		};
 	};
