diff -urN a/arch/arm64/boot/dts/mediatek/mt7981-spim-nand-gsw.dts b/arch/arm64/boot/dts/mediatek/mt7981-spim-nand-gsw.dts
--- a/arch/arm64/boot/dts/mediatek/mt7981-spim-nand-gsw.dts	2025-04-01 09:55:16.376719825 +0000
+++ b/arch/arm64/boot/dts/mediatek/mt7981-spim-nand-gsw.dts	2025-04-01 11:33:04.005210924 +0000
@@ -67,6 +67,16 @@
 				label = "ubi";
 				reg = <0x580000 0x6E00000>;
 			};
+
+			partition@7400000 {
+				label = "LogA";
+				reg = <0x7400000 0x800000>;
+			};
 		};
 	};
