diff -urN a/arch/arm64/boot/dts/mediatek/mt7981-spim-nand-gsw.dts b/arch/arm64/boot/dts/mediatek/mt7981-spim-nand-gsw.dts
--- a/arch/arm64/boot/dts/mediatek/mt7981-spim-nand-gsw.dts	2025-03-29 03:12:42.026810346 +0000
+++ b/arch/arm64/boot/dts/mediatek/mt7981-spim-nand-gsw.dts	2025-03-29 02:59:22.666999359 +0000
@@ -53,7 +53,7 @@
 				reg = <0x0100000 0x0080000>;
 			};
 
-			partition@180000 {
+			factory_part: partition@180000 {
 				label = "Factory";
 				reg = <0x180000 0x0200000>;
 			};
@@ -128,7 +128,7 @@
                 compatible = "mediatek,eth-mac";
                 reg = <0>;
                 phy-mode = "2500base-x";
-
+				mtd-mac-address = <&factory_part 0x2A>;
                 fixed-link {
                         speed = <2500>;
                         full-duplex;
@@ -141,6 +141,7 @@
                 reg = <1>;
                 phy-mode = "gmii";
                 phy-handle = <&phy0>;
+				mtd-mac-address = <&factory_part 0x24>;
         };
 
         mdio: mdio-bus {
