diff -Nur openwrt_240809/package/network/services/dnsmasq/files/check_link.sh openwrt_240830/package/network/services/dnsmasq/files/check_link.sh
--- openwrt_240809/package/network/services/dnsmasq/files/check_link.sh	2024-08-30 18:28:29.527493163 +0900
+++ openwrt_240830/package/network/services/dnsmasq/files/check_link.sh	2024-08-30 18:55:26.620469600 +0900
@@ -9,9 +9,12 @@
 fi
 
 CAMERA_CONF=/etc/config/camera
+CAMERA_CONF_DEFAULT=/etc/config/camera_default
 if [ -e $CAMERA_CONF ]; then
+		echo "Initialize 'Camera config'"
 		rm ${CAMERA_CONF}
-		touch ${CAMERA_CONF}
+		cp ${CAMERA_CONF_DEFAULT} ${CAMERA_CONF}
+		#touch ${CAMERA_CONF}
 		rm -rf /etc/cameras
 fi
 
