diff -urN a/swlib.c b/swlib.c
--- a/swlib.c	2024-06-24 16:52:02.986981512 +0900
+++ b/swlib.c	2024-07-30 11:18:31.237246324 +0900
@@ -348,6 +348,8 @@
 	if (!n)
 		goto nla_put_failure;
 
+	if (link->link)
+		NLA_PUT_FLAG(msg, SWITCH_LINK_FLAG_LINK);
 	if (link->duplex)
 		NLA_PUT_FLAG(msg, SWITCH_LINK_FLAG_DUPLEX);
 	if (link->aneg)
@@ -427,6 +429,7 @@
 	CMD_DUPLEX,
 	CMD_ANEG,
 	CMD_SPEED,
+	CMD_ADMIN,
 };
 
 int swlib_set_attr_string(struct switch_dev *dev, struct switch_attr *a, int port_vlan, const char *str)
@@ -494,6 +497,14 @@
 					cmd = CMD_ANEG;
 				else if (!strcmp(ptr, "speed"))
 					cmd = CMD_SPEED;
+				else if ((!strcmp(ptr, "down")) || (!strcmp(ptr, "0"))) {
+					cmd = CMD_ADMIN;
+					link->link = 0;
+				}
+				else if ((!strcmp(ptr, "up")) || (!strcmp(ptr, "1"))) {
+					cmd = CMD_ADMIN;
+					link->link = 1;
+				}
 				else
 					fprintf(stderr, "Unsupported option %s\n", ptr);
 				break;
@@ -519,6 +530,19 @@
 				link->speed = atoi(ptr);
 				cmd = CMD_NONE;
 				break;
+			case CMD_ADMIN:
+				if (!strcmp(ptr, "up"))
+					link->link = 1;
+				else if (!strcmp(ptr, "down"))
+					link->link = 0;
+				if (!strcmp(ptr, "1"))
+					link->link = 1;
+				else if (!strcmp(ptr, "0"))
+					link->link = 0;
+				else
+					fprintf(stderr, "Unsupported value %s\n", ptr);
+				cmd = CMD_NONE;
+				break;
 			}
 		}
 		val.value.link = link;
